### ISSUE TITLE
Can't get the first target to start when created from the admin dashboard

### ISSUE THREAD

**Issue by Infouzi** [echoCTF/echoCTF.RED#464]
Hi,

I have a problem when trying to "spin" the first target (from [https://echoctfred.readthedocs.io/DOCKER-COMPOSE/](url)), It loads for a certain amount of time then i get this message: "Failed to restart container. Connection timed out".
<img width="1172" alt="target-failed" src="https://user-images.githubusercontent.com/61884413/144394358-8f363167-04af-48c3-aa3d-5352f4663724.png">

Now, i know a lot of reasons could be the problem. However, i was doubting the **dockerd url** (which is visible in the bottom of the previous image), because when i change it to the one given in the docs, it doesn't start at all.

What i mean is: since i'm on mac, i used a little hack to get the dockerd available using a socat container, and i used the internal ip address of the socat container combined to it's ip (which naturally should point to my host dockerd).

I don't really know where the problem is, and i was wondering if something similar happened before to help me with it.

Thanks

Here is my running containers and available networks.
### Containers
<img width="1547" alt="containers-up" src="https://user-images.githubusercontent.com/61884413/144395007-792a8697-783f-4b87-8d17-66e08fdb9796.png">

### Networks
<img width="1014" alt="network-config" src="https://user-images.githubusercontent.com/61884413/144396042-552c221b-24ed-48ab-91c4-9fa7f4e1060c.png">

**Comment by proditis** [comment]
Hi there, 

These error messages are long overdue to be replaced with something more helpful and meaningful sorry you had the bad luck to see them :smile: 

The server field for the target `tcp://172.17.0.3:2375` needs to be accessible by the backend container. I am guessing the backend container is unable to connect to that IP.

Try and start the `socat` container connected to the `echoctfred_private` network (`--network=echoctfred_private`) and try to connect from the backend container with `nc` to ensure its accessible. 

Keep in mind it will have a different IP so note it and update the server field.


Let me know if this helped.

**Comment by Infouzi** [comment]
Hello @proditis ,

Really sorry for taking this long to answer, but i will be here from now on.

So, here's what i did:

- I connected my socat container to the `echoctfred_private` and it has an address from that network.
- I used `docker exec -it echoctfred_backend bash` to try and `nc` the socat container from within.
- nc was not recognized, so i `pinged` the socat container to check if it's there and it was successful.
- Then, i `curled` the socat endpoint like this: `curl 172.24.0.4:2375/images/json` (172.24.0.4 being the socat container address, and the port exposed matches the docker host api endpoint);
- The latest command was successfull and i received and answer back ==> It means that , **they're on the same network** AND the **endpoint is available from within the backend container**

At first, it didn't work, but then and as I finished writing this, i tried spinning again and... **TADAAAAA** , it started the container successfully.

So, i'm leaving this here for other people to try and debug the same way just in case.

Thank you very much, much appreciated. <3
